### PR TITLE
Add addons doc template

### PR DESCRIPTION
**Pull request by hossainemruz**
Tasks:
- [x] PostgreSQL
- [x] Elasticsearch
- [x] MySQL
- [x] MongoDB
- [x] Add addon doc overview in guides

### CODE REVIEW

**Comment by tamalsaha** [comment]
Do we want to use `stashed/catalog/raw/master/` master branch ?

File: docs/guides/latest/addons/overview.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+The job model is further divided into two categories. In the first category, the targeted resource is well known to Stash (example, a PVC). Hence, the Stash operator itself can create the required job to backup/restore the target. In the second category, Stash follows `Function-Task` model where the targeted resource is not known to Stash. In this case, the user creates some [Function](/docs/concepts/crds/function.md) which resembles a step of backup/restore process and a [Task](/docs/concepts/crds/task.md) which specifies the order of execution of these steps. Stash uses these `Function` and `Task` to generate the required job definition to backup/restore the target.
+
+The `Function-Task` model enables Stash to backup/restore the resources that the operator itself is not aware of. Users can extend Stash by creating respective `Function` and `Task` to backup their desired resources.
+
+A Stash addon is a collection of [Functions](/docs/concepts/crds/function.md) and a [Task](/docs/concepts/crds/task.md) to backup & restore a specific resource.
```
Comment (tamalsaha) [inline docs/guides/latest/addons/overview.md]: Should we also mention that they need to write the docker images?

File: docs/addons/mongodb/setup/install.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+The flowing flags are available for customizing MongoDB addon installation:
+
+| Flag                | Usage                                                                                          |
+| ------------------- | ---------------------------------------------------------------------------------------------- |
+| `--version`         | Specify a specific version of a specific addon to install. Use it along with `--catalog` flag. |
+| `--docker-registry` | Specify the docker registry to use to pull respective addon images. Default Value: `stashed`.  |
+| `--image`           | Specify the name of the docker image to use for respective addons.                             |
+| `--image-tag`       | Specify the tag of the docker image to use for respective addon.                               |
+| `--mg-backup-args`  | Specify optional arguments to pass to `mongodump` command during backup.                       |
```
Comment (tamalsaha) [inline docs/addons/mongodb/setup/install.md]: Why do we pass these args at the installer level? Shouldn't these be used at BackupConfiguration level?

File: docs/guides/latest/addons/overview.md (hunk 1 above)
Comment (hossainemruz) [reply docs/guides/latest/addons/overview.md]: done

**Pull request merged**